tests/test_sentinel.py

```diff
@@ -64,156 +84,129 @@ def timeout_if_down(self, node):
-        sentinel.discover_master('mymaster')
-
-
-def test_discover_slaves(cluster, sentinel):
-    assert sentinel.discover_slaves('mymaster') == []
-
-    cluster.slaves = [
-        {'ip': 'slave0', 'port': 1234, 'is_odown': False, 'is_sdown': False},
-        {'ip': 'slave1', 'port': 1234, 'is_odown': False, 'is_sdown': False},
-    ]
-    assert sentinel.discover_slaves('mymaster') == [
-        ('slave0', 1234), ('slave1', 1234)]
-
-    # slave0 -> ODOWN
-    cluster.slaves[0]['is_odown'] = True
-    assert sentinel.discover_slaves('mymaster') == [
-        ('slave1', 1234)]
-
-    # slave1 -> SDOWN
-    cluster.slaves[1]['is_sdown'] = True
-    assert sentinel.discover_slaves('mymaster') == []
-
-    cluster.slaves[0]['is_odown'] = False
-    cluster.slaves[1]['is_sdown'] = False
-
-    # node0 -> DOWN
-    cluster.nodes_down.add(('foo', 26379))
-    assert sentinel.discover_slaves('mymaster') == [
-        ('slave0', 1234), ('slave1', 1234)]
-    cluster.nodes_down.clear()
-
-    # node0 -> TIMEOUT
-    cluster.nodes_timeout.add(('foo', 26379))
-    assert sentinel.discover_slaves('mymaster') == [
-        ('slave0', 1234), ('slave1', 1234)]
-
-
-def test_master_for(cluster, sentinel, master_ip):
-    master = sentinel.master_for('mymaster', db=9)
-    assert master.ping()
-    assert master.connection_pool.master_address == (master_ip, 6379)
-
-    # Use internal connection check
-    master = sentinel.master_for('mymaster', db=9, check_connection=True)
-    assert master.ping()
-
-
-def test_slave_for(cluster, sentinel):
-    cluster.slaves = [
-        {'ip': '127.0.0.1', 'port': 6379,
-         'is_odown': False, 'is_sdown': False},
-    ]
-    slave = sentinel.slave_for('mymaster', db=9)
-    assert slave.ping()
-
-
-def test_slave_for_slave_not_found_error(cluster, sentinel):
-    cluster.master['is_odown'] = True
-    slave = sentinel.slave_for('mymaster', db=9)
-    with pytest.raises(SlaveNotFoundError):
-        slave.ping()
-
-
-def test_slave_round_robin(cluster, sentinel, master_ip):
-    cluster.slaves = [
-        {'ip': 'slave0', 'port': 6379, 'is_odown': False, 'is_sdown': False},
-        {'ip': 'slave1', 'port': 6379, 'is_odown': False, 'is_sdown': False},
-    ]
-    pool = SentinelConnectionPool('mymaster', sentinel)
-    rotator = pool.rotate_slaves()
-    assert next(rotator) in (('slave0', 6379), ('slave1', 6379))
-    assert next(rotator) in (('slave0', 6379), ('slave1', 6379))
-    # Fallback to master
-    assert next(rotator) == (master_ip, 6379)
-    with pytest.raises(SlaveNotFoundError):
-        next(rotator)
-
-
-def test_ckquorum(cluster, sentinel):
-    assert sentinel.sentinel_ckquorum("mymaster")
-
-
-def test_flushconfig(cluster, sentinel):
-    assert sentinel.sentinel_flushconfig()
-
-
-def test_reset(cluster, sentinel):
-    cluster.master['is_odown'] = True
-    assert sentinel.sentinel_reset('mymaster')
+    def test_discover_master(sentinel, master_ip):
```

Comment:
I  moved all tests to be under a class so I could use 'skip_if_cluster_mode' on all together.